derive/tests/macros.rs

```diff
@@ -222,6 +231,49 @@ fn should_accept_any_raw_params() {
 	assert_eq!(expected, result4);
 }
 
+#[test]
+fn should_accept_optional_param() {
+	let mut io = IoHandler::new();
+	let rpc = RpcImpl::default();
+	io.extend_with(rpc.to_delegate());
+
+	// when
+	let req1 = r#"{"jsonrpc":"2.0","id":1,"method":"sqr","params":[2]}"#;
+	let req2 = r#"{"jsonrpc":"2.0","id":1,"method":"sqr","params":[]}"#;
+	let req3 = r#"{"jsonrpc":"2.0","id":1,"method":"sqr","params":null}"#;
```

Comment:
> hrm, really? In the client code Params::None would rejected then?

I was checking the code and it might not be what we do now. AFAICT in case of optional parameters we parse `None` correctly as well. Anyways, I think this discussion is a bit tangential to that particular PR, so let's open an issue and discuss there.

The code in PR is an improvement for the client side for sure, we might want to follow up on how we want the server to behave separately.